### PR TITLE
chore(front): component page

### PR DESCRIPTION
**Pull request by OlivPi**

### CODE REVIEW

**Comment by lutangar** [comment]
> #6

**Review by Pipazoul — APPROVED** [review]

File: apps/frontend/src/style/customTheme.ts [hunk 1]
```diff
@@ -0,0 +1,198 @@
+import { themeLT } from 'kepler.gl/dist/styles';
```
Comment (lutangar) [inline apps/frontend/src/style/customTheme.ts:1]: theme.ts -> export -> theme

File: apps/frontend/src/components/KeplerMap.tsx [hunk 2]
```diff
@@ -1,11 +1,17 @@
 import React from 'react';
 import KeplerGl from 'kepler.gl';
 import { AutoSizer } from 'react-virtualized';
+import { datatlasTheme } from '../style/customTheme';
```
Comment (lutangar) [inline apps/frontend/src/components/KeplerMap.tsx:4]: File should probably be named `theme.ts` and export a `theme` const.

File: apps/frontend/src/components/badges/InfoProjectBadges.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  justify-content: center;
+  align-items: center;
+  width: 30px;
+  height: 30px;
+  border: 1.5px solid #000000;
+  border-radius: 50%;
+  margin-left: 10px;
+`;
+
+const InfoProjectBadges = ({ adminInitial, editorsNumber }) => {
```
Comment (lutangar) [inline apps/frontend/src/components/badges/InfoProjectBadges.tsx:22]: Props should be typed.

File: apps/frontend/src/components/buttons/IconButton.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React, { MouseEventHandler, ReactElement } from 'react';
+import { MapControlButton } from 'kepler.gl/dist/components/common/styled-components';
+
+interface IconButtonProps {
```
Comment (lutangar) [inline apps/frontend/src/components/buttons/IconButton.tsx:4]: :+1:

File: apps/frontend/src/components/buttons/IconTextButton.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React, { MouseEventHandler, ReactElement, ReactNode } from 'react';
+import { Button } from 'kepler.gl/dist/components/common/styled-components';
+
+interface IconButtonProps {
+  children: ReactNode;
+  Icon: ReactElement;
+  onClick: MouseEventHandler;
+}
+export const IconTextButton = ({ children, Icon, onClick }: IconButtonProps) => {
```
Comment (lutangar) [inline apps/frontend/src/components/buttons/IconTextButton.tsx:9]: This component should most likely simply be named `Button`...

File: apps/frontend/src/components/buttons/IconTextButton.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React, { MouseEventHandler, ReactElement, ReactNode } from 'react';
+import { Button } from 'kepler.gl/dist/components/common/styled-components';
+
+interface IconButtonProps {
```
Comment (lutangar) [inline apps/frontend/src/components/buttons/IconTextButton.tsx:4]: Interface name isn't consistent.

File: apps/frontend/src/components/card/CardDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import styled from 'styled-components';
+import InfoProjectBadges from '../badges/InfoProjectBadges';
+import CardContent from '../content/CardContent';
+
+interface CardContentProps {
```
Comment (lutangar) [inline apps/frontend/src/components/card/CardDetails.tsx:6]: Interface name isn't consistent.

File: apps/frontend/src/components/card/CardDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  }
+`;
+const ActionsCardContainer = styled.div`
+  display: flex;
+  list-style: none;
+  justify-content: space-around;
+  margin: 10px 0;
+`;
+
+const CardDetails = ({ titleCard, infoStatus, desc, adminInitial, editorsNumber }: CardContentProps) => {
```
Comment (lutangar) [inline apps/frontend/src/components/card/CardDetails.tsx:29]: If this component is only used for projects it should include `Project` in the name.

File: apps/frontend/src/components/card/CardDetails.tsx [hunk 9]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import styled from 'styled-components';
+import InfoProjectBadges from '../badges/InfoProjectBadges';
+import CardContent from '../content/CardContent';
+
+interface CardContentProps {
+  titleCard: string;
+  infoStatus?: string;
+  desc: string;
+  adminInitial: string;
```
Comment (lutangar) [inline apps/frontend/src/components/card/CardDetails.tsx:10]: These props shouldn't be re-declared here but you should reuse those of the `ProjectBadges` component instead.
> see `intersection-types` https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types

File: apps/frontend/src/components/card/MapPreview.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Trash, Delete, MapIcon } from 'kepler.gl/dist/components/common/icons';
+import { IconButton, IconTextButton } from '../buttons';
+
+interface MapPreviewProps {
+  status: boolean;
```
Comment (lutangar) [inline apps/frontend/src/components/card/MapPreview.tsx:7]: A boolean status property doesn't mean anything.
You should instead reuse definitions define earlier and now available in the code ! (see `src/shared/models`)
Also if you thought about the `published` property of a project, I made a last minute change in #54 an it's now renamed to `draft`.

File: apps/frontend/src/components/card/ProjectCard.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+// }
+
+const CardContainer = styled.article`
+  display: flex;
+  position: relative;
+  width: 50vw;
+  margin: auto;
+`;
+const ProjectCard = (props) => {
+  const { published, titleCard, infoStatus, desc, adminInitial, editorsNumber } = props;
```
Comment (lutangar) [inline apps/frontend/src/components/card/ProjectCard.tsx:23]: Props destructuration should be done in the components args whenever possible.

Comment (lutangar) [reply apps/frontend/src/components/card/ProjectCard.tsx:23]: Props should also be typed.

File: apps/frontend/src/components/card/ProjectCard.tsx [hunk 12]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import MapPreview from './MapPreview';
+import CardDetails from './CardDetails';
+import styled from 'styled-components';
+
+// interface TestProjectsProps {
```
Comment (lutangar) [inline apps/frontend/src/components/card/ProjectCard.tsx:6]: You should type the props instead.

File: apps/frontend/src/components/content/CardContent.tsx [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const StatusModif = styled.p`
```
Comment (lutangar) [inline apps/frontend/src/components/content/CardContent.tsx:4]: You should find a meaningful English name.

**Review by lutangar — CHANGES_REQUESTED** [review]
All components props should be typed and __you should be very careful to find meaningful naming for props and component__. Spend more time looking for name, use online dictionary, translator, chatgpt, or anyhting rly... but moreover reuse the __domain wording__ define together during the earlier design phase and now partially in the `src/shared/models`.

You should also spend some time re-organising the theme properties.

**Comment by lutangar** [comment]
Also, I shouldn't be the one to tell you that since your local hook AND the CI is telling you but... You should run the format command since the format of this file isn't correct :
`apps/frontend/src/components/card/MapPreview.tsx`.

**Comment by lutangar** [comment]
> This branch has conflicts that must be resolved

Rebase please.

File: apps/frontend/src/components/card/CardDetails.tsx (hunk 9 above)
Comment (OlivPi) [reply apps/frontend/src/components/card/CardDetails.tsx]: What do you think if declare my props on the `ProjectCard` component ? 
They include the props of the `InfoProjectBadges`. I don't have any errors but is this the right way to do it?

File: apps/frontend/src/components/footer/footer.tsx [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+  children: ReactNode;
+};
+
+const FooterContainer = styled.footer`
+  grid-area: footer;
+  padding: ${(props) => props.theme.layoutsBoxContainer};
+  height: 10vh;
+`;
+
+const Footer = (FooterProps) => <FooterContainer>Je suis le footer</FooterContainer>;
```
Comment (lutangar) [inline apps/frontend/src/components/footer/footer.tsx]: Props + type

File: apps/frontend/src/components/content/CardContent.tsx [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import styled from 'styled-components';
+
+const CardStatusInfos = styled.p`
+  color: #cecece;
+  margin: 0;
+`;
+const CardContent = ({ titleCard }) => {
```
Comment (lutangar) [inline apps/frontend/src/components/content/CardContent.tsx]: Type props

File: apps/frontend/src/components/sidebar/Sidebar.tsx [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+type SidebarProps = {
+  children: ReactNode;
+};
+
+const SideBarContainer = styled.aside`
+  grid-area: aside;
+  padding: ${(props) => props.theme.layoutsBoxContainer};
+`;
+
+const Sidebar = ({ children }) => (
```
Comment (lutangar) [inline apps/frontend/src/components/sidebar/Sidebar.tsx]: add type : SidebarProps

File: apps/frontend/src/components/card/CardDetails.tsx [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+  }
+`;
+const ActionsCardContainer = styled.div`
+  display: flex;
+  list-style: none;
+  justify-content: space-around;
+  margin: 10px 0;
+`;
+
+const CardDetails = ({ name }) => {
```
Comment (lutangar) [inline apps/frontend/src/components/card/CardDetails.tsx]: Type props

File: apps/frontend/src/components/card/MapPreview.tsx [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+  background-image: url('https://images.unsplash.com/photo-1569336415962-a4bd9f69cd83?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1931&q=80');
+`;
+
+const ActionsContainer = styled.div`
+  display: flex;
+  align-self: end;
+  height: 40px;
+`;
+
+const MapPreview = ({ published }) => {
```
Comment (lutangar) [inline apps/frontend/src/components/card/MapPreview.tsx]: Type props

File: apps/frontend/src/components/card/ProjectCard.tsx [hunk 19]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import MapPreview from './MapPreview';
+import CardDetails from './CardDetails';
+import styled from 'styled-components';
+import { DatasetInterface, NormalizedProjectInterface } from '@datatlas/shared/models';
+
+// interface CardProps {
```
Comment (lutangar) [inline apps/frontend/src/components/card/ProjectCard.tsx]: Remove

File: apps/frontend/src/components/card/ProjectCard.tsx [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+//   draft: boolean;
+//   titleCard: string;
+//   infoStatus?: string;
+//   desc: string;
+//   adminInitial: string;
+//   editorsNumber: number;
+//   projects: DatasetInterface[];
+// }
+export type ProjectListItemProps = NormalizedProjectInterface;
```
Comment (lutangar) [inline apps/frontend/src/components/card/ProjectCard.tsx]: Rename to match component name

File: apps/frontend/src/index.tsx [hunk 21]
```diff
@@ -18,6 +19,7 @@ const router = createBrowserRouter([
     errorElement: <ErrorPage />,
     children: [
       {
+        path: 'projects',
```
Comment (lutangar) [inline apps/frontend/src/index.tsx]: Remove.

File: apps/frontend/src/pages/ProjectsPage.tsx [hunk 22]
```diff
@@ -1,11 +1,45 @@
...
 import { useGetSavedProjectsQuery } from '../api';
 import { StartNewProjectForm } from '../components/forms/StartNewProjectForm';
 import { startNewProject } from '../store/reducers/app/drafts';
 import { selectCurrentUserProjects } from '../store/selectors';
 import { useAppDispatch } from '../store';
+import Footer from '../components/footer/footer';
+import Sidebar from '../components/sidebar/Sidebar';
+import Navbar from '../components/nav/Navbar';
+
+const testProjectsDatas = {
```
Comment (lutangar) [inline apps/frontend/src/pages/ProjectsPage.tsx]: You could use generateProject instead and should be typed.

File: apps/frontend/src/pages/ProjectsPage.tsx [hunk 23]
```diff
@@ -1,11 +1,45 @@
...
+
+const LayoutProjects = styled.div`
+  display: grid;
+  height: 100vh;
+  grid-template-rows: 0.2fr 1fr 0.5fr 0.5fr;
+  grid-template-areas:
+    'nav nav nav nav'
+    'main main main aside'
+    'footer footer footer footer';
+  //text-align: center;
```
Comment (lutangar) [inline apps/frontend/src/pages/ProjectsPage.tsx]: comment?

File: apps/frontend/src/components/ProjectList.tsx [hunk 24]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
-import { ProjectListItem } from './ProjectListItem';
+// import { ProjectListItem } from './ProjectListItem';
```
Comment (lutangar) [inline apps/frontend/src/components/ProjectList.tsx]: Leftover comment.

**Pull request merged**